### PR TITLE
Remove while enumerating dictionary

### PR DESCRIPTION
**Pull request by danmosemsft**
Fix https://github.com/dotnet/corefx/issues/31459

There were surprisingly few instances of this pattern found by @Wraith2 's analyzer that were not false positives.

### CODE REVIEW

File: src/System.Data.SqlClient/src/System/Data/SqlClient/SqlDependencyUtils.cs [hunk 1]
```diff
@@ -324,9 +324,6 @@ private void RemoveDependencyFromCommandToDependenciesHash(SqlDependency depende
         {
             lock (_instanceLock)
             {
-                List<string> notificationIdsToRemove = new List<string>();
```
Comment (jkotas) [inline src/System.Data.SqlClient/src/System/Data/SqlClient/SqlDependencyUtils.cs]: System.Data.SqlClient builds downlevel versions. This won't work downlevel.

**Comment by geoffkizer** [comment]
HttpClient changes look good to me.

File: src/System.Data.SqlClient/src/System/Data/SqlClient/SqlDependencyUtils.cs (hunk 1 above)
Comment (jkotas) [reply src/System.Data.SqlClient/src/System/Data/SqlClient/SqlDependencyUtils.cs]: >  guess I need to wait for this change to reach ProjectN.

It is in ProjectN

Comment (danmosemsft) [reply src/System.Data.SqlClient/src/System/Data/SqlClient/SqlDependencyUtils.cs]: OK. And for SqlClient, from talking to @joperezr , it is not necessarily building too much. I will leave it as is.

**Comment by danmosemsft** [comment]
Only failure is in drawing.

**Pull request merged**